api/controller/base.py

```diff
@@ -120,4 +120,16 @@ def library_for_request(self, library_short_name):
         if not library:
             return LIBRARY_NOT_FOUND
         flask.request.library = library
+
+        # Finland: For authenticated patrons, override the library defined in the request
+        # and use the library assigned to that patron.
+        #
+        # This is a bit sketchy as the authentication requires library to be set already.
```

Comment:
Any risk to get duplicate records?

Reply:
I don't think duplicate records are a risk here. The risk is I might've missed some detail that would make this break in some specific cases. Pretty sure it's fine though. I suppose selecting the library (automatically) on client side might be a bit safer as the requests would then use the default (upstream) logic.